src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Mime;
+using System.Text;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Service.Configurations;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Writers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Services.OpenAPI
+{
+    /// <summary>
+    /// Service which generates and provides an OpenAPI description document
+    /// describing the DAB engine's REST endpoint paths.
+    /// </summary>
+    public class OpenApiDocumentor : IOpenApiDocumentor
+    {
+        private ISqlMetadataProvider _metadataProvider;
```

Comment:
nit: we have been using `private static readonly` for all service dependencies

Reply:
I'm updating to `private readonly`. I did not see examples/reason to use static here (static constructor would also prevent usage of constructor injection of services).https://stackoverflow.com/questions/55213803/use-dependency-injection-in-static-class